### DISCUSSION TITLE
The right way to update NetAlertX ?

### DISCUSSION THREAD

**Discussion by AdmiralRaccoon** [netalertx/NetAlertX#670] [General]
Hi there, 
this is maybe not directly NetAlert related, but I´m not sure if I´m doing the update the correct way, as I had some issues after the last updates. Luckily, it´s an VM and made a complete backup and restored it.

However, I´m quite a docker noob. For the last updates, I simply went within Portainer to the NetAlertX Stack-Editor and repulled the image. (See pics). Is that the correct way?

Thank you and a nice weekend.
Philipp

![update1](https://github.com/jokob-sk/NetAlertX/assets/41727850/123203ad-a3d3-4d84-9157-41fd3a6b43ee)
![image](https://github.com/jokob-sk/NetAlertX/assets/41727850/53c4e2b6-29cf-4c5a-91a4-02bfd8c5e1ce)

**Comment by jokob-sk** [comment]
Hi @AdmiralRaccoon (love the username) !

I checked my Portainer instance, but I can't seem to find the dialog from your screenshot. 

Just to be sure, if you are moving from PiAlert to NetAlertX, double-check the migration guide: https://github.com/jokob-sk/NetAlertX/blob/main/docs/MIGRATION.md

Especially backup the files (`pialert.db`, `pialert.conf`, `devices.csv`) mentioned in the guide so you don't lose your configuration if something goes wrong. If you then rename the original (or backed-up files `pialert.db` renamed to `app.db`, `pialert.conf` renamed to `app.conf`) in the right location (mapped to `/app/db` and `/app/conf` ), then everything should work fine.

If you already performed the migration to NetAlertX then the above is unnecessary and the portainer approach should work fine.  

Let me know if you have more questions,
j

**Reply by masterwishx** [reply]
Updated in Unraid all fine, but we should rename files also for load the all configs

**Comment by vmsman** [comment]
Everytime I have upgraded NetalertX, I end up having to restore the database and restore all of the nodes.  Even when I do so, the system forgets all of my settings for alerts, etc.  It's a very manual process.  I have nearly 200 systems on my Homelab and I have all but decided to never take another upgrade.

**Reply by jokob-sk** [reply]
Hi @vmsman ,

Sorry to hear that. That's definitely not what should happen. Could you please describe your setup?

- What is your docker compose file? 
- What plugins do you have enabled?
- Why do you have to restore the DB / nodes?
   - are there any errors in the log or in the browser console?
   - are you clearing the cache in the browser and in the app?
   
Thanks in advance for the details and I hope I can help.

**Reply by vmsman** [reply]
My docker-compose:

```
services:
  netalertx:
    container_name: netalertx
    image: "jokobsk/netalertx:latest"
    network_mode: "host"
    restart: unless-stopped
    volumes:
      - ./config:/app/config
      - ./db:/app/d
    environment:
      - TZ=America/Chicago
      - PORT=80
```

My plugins are Arp-scan, Internet-check, AVIHISCAN, NSLookup, Pholus, and the Apprise publisher.
The only things in the logs are [NSLOOKUP]No PTR record found for IP: 172.16.1.221 and many other addresses.  Those addresses are in the database and tracked just fine.
I have to do a restore because after every update (at least the last three) I have had to do a restore because only 1 My Device and 1 Connected device and everything has to be discovered from scratch.  I do a CSV import to bring everything back.  The only thing not broken after an update are my notification settings for Apprise.  
Yes, I am clearing the cache in the browser.

The huge downside of restoring from the CSV is that I have to edit all 200 of the entries to reset some of the settings which are lost and then the program ends up marking all the nodes as new anyway and all my favorites are also gone and need to be reentered.  Updates are scary and super labor intensive.

**Reply by jokob-sk** [reply]
Hummm, if the above docker-compose.yml is really the one you use then you are not mapping the correct db path. It should be `/app/db` not `/app/d`. See the docs for more details:

https://github.com/jokob-sk/NetAlertX/blob/main/dockerfiles/README.md#docker-paths

**Reply by vmsman** [reply]
Wow, gee thanks.  That was in fact a typo and one that has existed for quite some time apparently.  Fixed now.  Perhaps upgrades will be less painful now.  Also, whenever I update any setting on a single node it seems to stay in the "loading..." operation for 2-3 minutes.  Is that normal?

**Reply by jokob-sk** [reply]
Fingers crossed! The save operation depends on the number of enabled plugins and HW that you are using. I have a Synology 920 w an ssd and for 12 plugins it takes ~10s to save and reload the application.

**Reply by vmsman** [reply]
My NetAlertX is hosted in an Incus container and it has plenty of resources.  The host is a Ryzen 5.  The last version the save operation took about 20 seconds.  In this version the save operation for any one device takes 2-3 minutes.

**Reply by jokob-sk** [reply]
Hummm, that does not sound right. Can you please  open a separate issue and I can have a closer look. By filling in the issue template it will help me get enough information from the get go.

Thanks,
J

**Reply by jokob-sk** [reply]
Oh, also, please check this guide on how to make sure the required maintenance tasks are running as they impact performance significantly: https://github.com/jokob-sk/NetAlertX/blob/main/docs/PERFORMANCE.md

**Reply by vmsman** [reply]
@jokob-sk After a lot of additional testing, I believe I found the issue.  Apparently my typo that you pointed out on the container database was responsible for some type of corruption.  I had to stop the container explicitly and change the docker compose to fix the typo and then restart the docker compose.  This of course nuked the node listing which was predictable due to no actual connection to the database.  I then loaded my exported node list back in.  Updatas still ran very slowly, but for only about an hour.  At this point, any node edits that I make are immediate without the aforementioned long delay.  

There is only one remaining mystery.  I have an Intel Atom server with a connection to enp4s0 (i226 device) and device eno1 (X553 10G fiber SFP+).  Device enp4s0 has address 172.16.1.77.  Device eno1 is not assigned an address but is used as the parent for a software bridge I named bridge0 and it has address 172.16.1.82.  NetalertX reports two to three times per day that this system has changed addresses and yet it has not because both aforementioned addresses are DHCP address reservations on the router.  NetAlertX is sending notifications that the addresses have changed.  Yet, the old and new addresses are the same as are the MAC addresses.  The only thing that I can come up is that NetalertX might be seeing the bridge address and then the physical eno1 device thinking that is a change.  Your thoughts?

![image](https://github.com/user-attachments/assets/b33634e8-bbed-4fdc-b995-251974c807f4)

**Reply by jokob-sk** [reply]
Glad to hear that! 

We are getting off topic - which makes then others find and resolve similar issues more difficult - I'm also really not a network expert so I'd recommend to open a separate discussion on this issue as others might be able to help, or try to ask on the discord channel (not too active but people might respond). Alternatively, if you think its an app issue, please submit a new issue with the appropriate logs of when the notifications occur, probably in the trace log_level. 

https://github.com/jokob-sk/NetAlertX/blob/main/docs/DEBUG_PLUGINS.md
https://github.com/jokob-sk/NetAlertX/blob/main/docs/DEBUG_TIPS.md
